App.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { useState, useEffect } from "react";
+import {
+  View,
+  Text,
+  TouchableOpacity,
+  Platform,
+  StyleSheet,
+} from "react-native";
+import NfcManager, { NfcTech } from "react-native-nfc-manager";
+import { Buffer } from "buffer";
+
+const App: React.FunctionComponent = () => {
+  const [adpuResponse, setADPUResponse] = useState("APDU Response");
+  const [detectedTechnology, setDetectedTechnology] = useState("");
+  const [nfcUID, setNfcUID] = useState("");
+  const [testSuccess, setTestSuccess] = useState(false);
+  const [waitingCIE, setWaitingCIE] = useState(false);
+
+  const cleanUp = () => {
+    // Do the cleanings in any case
+    NfcManager.cancelTechnologyRequest().catch(() => 0);
+
+    // Reset state
+    setADPUResponse("APDU Response");
+    setDetectedTechnology("");
+    setNfcUID("");
+    setTestSuccess(false);
+    setWaitingCIE(false);
+  };
+
+  useEffect(() => {
+    NfcManager.start();
+    // Do the cleanings after rendering only if the app is not waiting for CIE
+    return () =>
```

Comment:
```suggestion
    if (!waitingCIE) {
      return cleanUp();
    }
    return;
```

Returning a Promise causes an error at compile time, maybe this may result cleaner, what do you think?

Reply:
To be honest I can't see your point. `useEffect` returns a function not a promise, indeed I can't reproduce the compile time error you are referring to. Obviously, the returned (anonymous) functions tests the waitingCIE status flag and optionally returns a promise... I thought that was an admissible behavior. I'll try to make the code more clear in the next commit